addons/im_livechat/models/discuss_channel.py

```diff
@@ -107,6 +107,30 @@ def _execute_command_help_message_extra(self):
     def execute_command_history(self, **kwargs):
         self.env['bus.bus']._sendone(self, 'im_livechat.history_command', {'id': self.id})
 
+    def execute_command_bot(self, **kwargs):
+        chatbot_script = self.env['chatbot.script'].browse(kwargs.get('id'))
+        if not self or not chatbot_script.exists():
+            return None
+        add_chatbot_as_member = True
+        if (len(self.channel_member_ids) > 2):
+            if (self.livechat_operator_id.id == chatbot_script.operator_partner_id.id):
+                add_chatbot_as_member = False
+            else:
+                self.sudo()._action_unfollow(self.livechat_operator_id, post_left_message=False)
```

Comment:
Hi @tsm-odoo, 
I believe that the operator could leave the conversation when he considers it necessary, i.e. when the new operator is assigned to the discussion.
We could open the new bot in a separate chat window, but this could be confusing for the visitor. In addition, the operator might still want to jump into the discussion if he wants to add information.

Reply:
Thanks for your answer, then in order to do so we should always remove the operator from the channel: he will keep receiving the messages in the first place since the thread is still open thus he will be considered as a guest (as per the specification of the "live chat history in discuss" task) but will be able to leave when he wants to (leaving the channel will remove its subscription from this channel which is not the case if he still is member).

Reply:
So, if the operator is removed then he won't be able to send any message in the chat only receive them. But, what if the bot was supposed to take some data from the visitor like a form and then exit and the `forward_operator` step was not configured. Then, how will the operator send any message if he needed to?

Reply:
When an operator views a channel as a guest and performs an action that would require membership, they are automatically added as a member, so that's not an issue. I'm more concerned about this inconsistency:

- If the user comes from a chat bot (members > 2), the operator is removed from the channel members.
- If not, they stay in the conversation.

I suppose this is another question for @paju-odoo: Do we want the operator to be notified each time a message comes up after a bot has started? If not, an operator who closes a chat will not be able to return once the chat bot takes the lead.

- If yes: We probably need to adapt the _process_step_forward_operator method so that we don't add two operators for the same live chat.
- If not: We probably should remove the operator from the members to allow them to leave whenever they want.

Anyway, I feel like the behavior should be the same whether the user comes from a bot or a classic chat.

Reply:
So, we keep the operator as a member of the channel, even when the new bot is launched. If a new operator is assigned to the channel, the initial operator can still decide to manually leave the channel so it's not an issue...